### DISCUSSION TITLE
KDE Classic (3.5.x/Trinity DE) theme

### DISCUSSION THREAD

**Discussion by AzumaHazuki** [lxqt/lxqt#2405] [Show and tell]
This is something I've been working on for ages now, and it's *finally* ready :) I've included a palette and a custom Openbox theme that does the best it can to get close to how the Plastik windeco used to look. Source available <a href=https://github.com/AzumaHazuki/lxqt-themes-kde-classic>here</a> as always. Please let me know if you find bugs or have suggestions!

Screenshots:
![Openbox-KDE-Classic](https://user-images.githubusercontent.com/67122280/230796598-5e03af1a-7cec-477b-92c4-69f114586ce8.png)
![LXQt-KDE-Classic](https://user-images.githubusercontent.com/67122280/230796600-7f10f57e-0a6e-4f00-b8a1-27560396d2a0.png)

**Comment by edwardp3** [comment]
It looks good, but there is an issue with the Keyboard State Indicator. At first, the C N S is bolded, as if all three are turned on and pressing any of the CAPS LOCK/NUM LOCK/SCROLL LOCK keys doesn't change the appearance of the corresponding letter.

![lxqt-kb-kde-theme](https://user-images.githubusercontent.com/73129450/232252305-1aa217bb-af8e-497a-a1a5-569fa265e677.png)

One drawback to using light LXQt themes in Debian, is due to the Bluetooth, Keyboard layout, notification, and volume control icons are white and look invisble with a light theme:

![lxqt-kde-panel-icons](https://user-images.githubusercontent.com/73129450/232252466-82c69950-b8e8-4edc-9579-41209fcbdc0e.png)

The Keyboard State Indicator is also behaving the same way with the three Xfce 4.x themes.

**Reply by AzumaHazuki** [reply]
Weird, it's working fine on my system...it starts in normal weight and goes bold on click. I have no idea what's going on there. Try killing and restarting the panel?

The relevant .qss snippet from the file is as follows:

```
#KbIndicator #CapsLockLabel {color: #8d8d8d; font: normal;}
#KbIndicator #CapsLockLabel:enabled {color: #000000; font: bold;}
#KbIndicator #CapsLockLabel:!enabled {color: #000000; font: normal;}
#KbIndicator #CapsLockLabel:disabled {color: #8d8d8d; font: normal;}

#KbIndicator #NumLockLabel {color: #8d8d8d; font: normal;}
#KbIndicator #NumLockLabel:enabled {color: #000000; font: bold;}
#KbIndicator #NumLockLabel:!enabled {color: #000000; font: normal;}
#KbIndicator #NumLockLabel:disabled {color: #8d8d8d; font: normal;}

#KbIndicator #ScrollLockLabel {color: #8d8d8d; font: normal;}
#KbIndicator #ScrollLockLabel:enabled {color: #000000; font: bold;}
#KbIndicator #ScrollLockLabel:!enabled {color: #000000; font: normal;}
#KbIndicator #ScrollLockLabel:disabled {color: #8d8d8d; font: normal;}
```

**Reply by edwardp3** [reply]
When I extract any of these themes, they appear to include additional folders, beyond what I am used to seeing. 

I would extract first, then copy the KDE-Classic folder from /themes, into ~/.local/lxqt/themes, But the additional folders: palettes, wallpapers, KDE-Classic.tar.xz? Are anything done with these and if so, where should they be placed?

**Reply by AzumaHazuki** [reply]
The contents of the palettes and wallpapers folders go into ~/.local/share/lxqt/palettes|wallpapers respectively, and the tarball gets decompressed into .themes, as it's the Openbox theme seen in the screenshots.

I actually juuuuust added one more line into the main theme file to correct an issue with the menu width sometimes being wrong.

**Reply by edwardp3** [reply]
Thanks.

I have KDE-Classic/Trinity working. As for the Keyboard Status, if in Configure Panel/Styling, Font Color is checked (defaults to black), that causes the C N S to display bolded, as shown above. With Font Color not checked, the non-active letters (C and S) are greyed out and the N is correctly bolded.

**Comment by isf63** [comment]
What Qt widgets style is used in these screenshots? I noticed the scrollbars have a different (more classic) look compared to Breeze and Fusion.

**Reply by AzumaHazuki** [reply]
Believe it or not, I did the scrollbars, sliders, and up/down arrow boxes custom. The Plastik QT style would do something similar but it would have up *and* down arrow-buttons at the bottom of the scrollbar, which always kind of made my brain itch...

The .qss directives in `lxqt-runner.qss` that produce the look are as follows:
```
QScrollBar::handle {
        color:                          transparent;
        background:                     qlineargradient(x1:0, y1:0, x2:1, y2:0, stop:0 #e8e9ef, stop:1 #cecfd5);

        background-image:               url(assets/scrollbars/dots.png);
        background-repeat:              no-repeat;
        background-position:            center;
        background-attachment:          fixed;

        border:                         2px solid transparent;
        border-image:                   url(assets/scrollbars/handle.png) 2;

        margin:                         16px 0px;

        width:                          16px;
}
```
